src/Components/CommentAdder/CommentAdder.jsx

```diff
@@ -0,0 +1,55 @@
+import { postComment } from "../../Utils/api";
+import { useState } from "react";
+import { useParams } from "react-router";
+import '../CommentAdder/CommentAdder.css'
+
+function CommentAdder({ setArticleComments, currentUser, setFetchedComments }) {
+  const { username } = currentUser;
+  const [newComment, setNewComment] = useState("");
+  const [commentMessage, setCommentMessage] = useState("");
+  const { article_id } = useParams();
+
+  function handleSubmit(event) {
+    setFetchedComments(false);
+    event.preventDefault();
```

Comment:
Optional suggestion - you could consider adding the comment to the comments array optimistically here!